### PR TITLE
[stabel/rocketchat] Improve chart

### PR DESCRIPTION
**Pull request by srueg**
#### What this PR does / why we need it:

Overall improvement and stabilization for the Rocket.Chat chart

#### Special notes for your reviewer:
Unfortunately the Rocket.Chat v1.x release requires MongoDB to run a replicaset and have access to the OpLog. I couldn't get it to work with the MongoDB Helm chart (authentication issues for the oplog), so I'll stick to 0.74.3 for now.
RocketChat/Rocket.Chat#14353

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

**Comment by srueg** [comment]
/assign @verwilst 
Please take a look at this. I implemented some improvements, any feedback is much appreciated.

**Comment by srueg** [comment]
Unfortunately the Rocket.Chat v1.x release requires MongoDB to run a replicaset and have access to the OpLog. I couldn't get it to work with the MongoDB Helm chart (authentication issues for the oplog), so I'll stick to `0.74.3` for now.
https://github.com/RocketChat/Rocket.Chat/issues/14353

File: stable/rocketchat/OWNERS [hunk 1]
```diff
@@ -1,4 +1,6 @@
 approvers:
 - verwilst
+- srueg
```
Comment (verwilst) [inline stable/rocketchat/OWNERS:3]: Does this mean you will keep on developing/maintaining this chart, or was this a one-time effort?

File: stable/rocketchat/README.md [hunk 2]
```diff
@@ -2,25 +2,114 @@
...
+`image.pullPolicy` | Image pull policy | `IfNotPresent`
+`host` | Hostname for Rocket.Chat. Also used for ingress (if enabled) | `""`
+`replicaCount` | Number of replicas to run | `1`
+`smtp.enabled` | Enable SMTP for sending mails | `false`
+`smtp.username` | Username of the SMTP account | `""`
+`smtp.password` | Password of the SMTP account | `""`
+`smtp.host` | Hostname of the SMTP server | `""`
+`smtp.port` | Port of the SMTP server | `587`
+`extraEnv` | Extra environment variables for Rocket.Chat. Used with `tpl` function, so this needs to be a string | `""`
+`existingMongodbURL` | MongoDB URL if using an externally provisioned MongoDB | `""`
```
Comment (verwilst) [inline stable/rocketchat/README.md]: I would call it externalMongodbUrl. You're not even sure it exists already. :-)

**Comment by verwilst** [comment]
Seems like these changes broke tests:
```
I0513 07:33:07.094] ==> Linting stable/rocketchat
I0513 07:33:07.094] Lint OK
I0513 07:33:07.094] 
I0513 07:33:07.094] 1 chart(s) linted, no failures
I0513 07:33:07.100] 2019/05/13 07:33:07 [INFO] Missing required value: Please specify a MongoDB password
I0513 07:33:07.100] 2019/05/13 07:33:07 [INFO] Missing required value: Please specify a MongoDB password
I0513 07:33:07.101] Installing chart 'rocketchat => (version: "1.0.0", path: "stable/rocketchat")'...
I0513 07:33:07.894] Error: render error in "rocketchat/templates/secret.yaml": template: rocketchat/templates/secret.yaml:20:19: executing "rocketchat/templates/secret.yaml" at <required "Please spe...>: error calling required: Please specify a MongoDB password
```

File: stable/rocketchat/OWNERS (hunk 1 above)
Comment (srueg) [reply stable/rocketchat/OWNERS:3]: Yes, I'll keep maintaining it. If that's ok with you.

**Comment by srueg** [comment]
Unfortunately I had to set a default password for the MongoDB as I don't see another way to pass the tests.

**Comment by cpanato** [comment]
You need to create a ci values and there add the values that you need to set. Please do not set hardcoded pass in the values.yaml

**Comment by srueg** [comment]
Ah thanks, I didn't know about `ci-values.yaml`. I updated the PR.

File: stable/rocketchat/templates/deployment.yaml [hunk 3]
```diff
@@ -3,25 +3,31 @@ kind: Deployment
...
+        checksum/secret: {{ include (print .Template.BasePath "/secret.yaml") . | sha256sum }}
     spec:
       {{- if .Values.securityContext.enabled }}
       securityContext:
         fsGroup: {{ .Values.securityContext.fsGroup }}
         runAsUser: {{ .Values.securityContext.runAsUser }}
       {{- end }}
       containers:
-      - name: {{ template "rocketchat.fullname" . }}
+      - name: rocketchat
```
Comment (verwilst) [inline stable/rocketchat/templates/deployment.yaml:30]: wouldn't this create duplicate containers if you install 2 rocketchat instances in the same namespace? Is there a reason why this is changed?

Comment (srueg) [reply stable/rocketchat/templates/deployment.yaml:30]: Yes, but a container name only needs to be unique inside the same pod. And it makes it easier if the name is the same for all releases (f.e. for things like `kubectl logs --container rocketchat`)

**Pull request merged**